### ISSUE TITLE
Natural / Integer syntax suggestion

### ISSUE THREAD

**Issue by sellout** [dhall-lang/dhall-lang#138]
tl;dr: Change the grammar to move the `+` sigil to Integers and Doubles, like 
```
Natural <- [0-9]+
Integer <- [+-] [0-9]+
Double <- [+-] [0-9]* '.' [0-9]+
```

I have a few justifications for this:
- personal: Using Dhall at work, we tend to mostly use `Natural`, with `Integer` and `Double` being extremely rare (I find this true of coding in Haskell, as well – it’s mostly the lack of availability of Naturals that makes it not the case in other languages, IMO, although I know it does also depend on the domains that one works in)
- mnemonics: “signed” vs “unsigned” numbers, also “natural” in the sense of “unadorned”
- syntax: Since `Natural` is the only numeric type with operations, so `+2 + +3` is, to me, much harder to read than `2 + 3`
- cardinality(?): I think encouraging smaller types is good practice, and simpler syntax for Naturals than Integers should help with that

I know it’s also a significant breaking change across like every config file ever, so …

**Comment by Gabriel439** [comment]
Yeah, it's a pretty breaking change, but I still think it's a good idea and I agree with all the points you made.  I think it's mainly a matter of giving this issue more visibility so that people can chime in about whether or not they think the disruption is worth the benefit.  I'll try to broadcast this more widely.

**Comment by joneshf** [comment]
I 100% support this!

**Comment by KirinDave** [comment]
I think the cleanup to arithmetic is worth it.

**Comment by vmchale** [comment]
@sellout I would note that the cardinality is the same in either case.

**Comment by sellout** [comment]
@vmchale Yeah, that‘s why I added the “(?)” 😄 I just didn’t have a better term for it.

**Comment by evincarofautumn** [comment]
Sounds like a good idea. A small conversion utility might be helpful for upgrading. I dunno how many users Dhall has at the moment, but it’s a very new language, and IMO people should expect some breaking changes while the core language is being figured out.

**Comment by f-f** [comment]
I totally agree with all of this thread 👍

**Comment by Gabriel439** [comment]
Seems like unanimous agreement to me.  If nobody objects by tomorrow then we should proceed with a pull request to change the semantics.

@sellout: Did you want to submit the pull request?  If not, I can do so
